include/seqan3/core/configuration/detail/concept.hpp

```diff
@@ -0,0 +1,116 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides concepts for the configuration classes.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <functional>
+#include <type_traits>
+
+#include <meta/meta.hpp>
+
+#include <seqan3/core/type_traits/template_inspection.hpp>
+#include <seqan3/std/concepts>
```

Comment:
Please move this header up to the std-includes and sort it as if its name is `<concepts>`